src/pods/movies/components/movies-filter.component.tsx

```diff
@@ -1,4 +1,134 @@
 import * as React from 'react';
+import { TextField, MenuItem, createStyles, WithStyles, Button } from '@material-ui/core';
+import { withStyles } from '@material-ui/core/styles';
+import { MovieFilter, GenresList } from '../view-model';
+import { GenreEntity } from '../../../api/model';
+import { genresAPI } from '../../../api/genres-api'
 
-export const MoviesFilterComponent = () => 
-  <h3>Hello from Movies MoviesFilterComponent</h3>
\ No newline at end of file
+const styles = theme => createStyles({
+  container: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    flexDirection: 'column',
+    marginRight: '30px',
+  },
+  textField: {
+    marginLeft: theme.spacing.unit,
+    marginRight: theme.spacing.unit,
+    marginBottom: theme.spacing.unit,
+    width: 300,
+  },
+  dense: {
+    marginTop: 19,
+  },
+  menu: {
+    width: 300,
+  },
+  button: {
+    margin: theme.spacing.unit,
+    width: 150,
+  },
+  input: {
+    display: 'none',
+  },
+});
+
+interface Props extends WithStyles<typeof styles> {
+  onApplyFilter: (movieFilter: MovieFilter) => void;
+}
+
+interface State {
+  movieFilter: MovieFilter;
+  genres: GenreEntity[];
+}
+
+export class MoviesFilterComponentInner extends React.Component<Props, State> {
+
+  state: State = {
+    movieFilter: {
+      title: '',
+      genre: '',
+      year: 2018,
```

Comment:
are we sure that we want this value by default? first loading doesn't apply any filter, and we are implementing to type filters, first apply all filters, but the second one applies only the filter which has changed.